test/e2e/unmanaged_test.go

```diff
@@ -11,54 +21,107 @@ import (
+//	err := patchAndCheckConfigMap(t, client, false)
+//	if err != nil {
+//		t.Fatalf("error: %s", err)
+//	}
+//}
+//
+//func TestEditUnmanagedService(t *testing.T) {
+//	client := testframework.MustNewClientset(t, nil)
+//	defer testframework.MustManageConsole(t, client)
+//	testframework.MustUnmanageConsole(t, client)
+//
+//	err := patchAndCheckService(t, client, false)
+//	if err != nil {
+//		t.Fatalf("error: %s", err)
+//	}
+//}
+//
+//func TestEditUnmanagedRoute(t *testing.T) {
+//	client := testframework.MustNewClientset(t, nil)
+//	defer testframework.MustManageConsole(t, client)
+//	testframework.MustUnmanageConsole(t, client)
+//
+//	err := patchAndCheckRoute(t, client, false)
+//	if err != nil {
+//		t.Fatalf("error: %s", err)
+//	}
+//}
 
-func TestEditUnmanagedConfigMap(t *testing.T) {
+// TEMP:  fiddling with the tests to see why things are failing in CI
+func TestEditUnmanagedConfigMapAgain(t *testing.T) {
+	frequency := 1 * time.Second
+	timeout := 10 * time.Second
 	client := testframework.MustNewClientset(t, nil)
+	// in the end, put the console back to "Manged"
 	defer testframework.MustManageConsole(t, client)
+	// kick us off with "Unmanaged"
 	testframework.MustUnmanageConsole(t, client)
 
-	err := patchAndCheckConfigMap(t, client, false)
+	originalCM, err := client.ConfigMaps(api.TargetNamespace).Get(api.OpenShiftConsoleConfigMapName, metav1.GetOptions{})
 	if err != nil {
 		t.Fatalf("error: %s", err)
 	}
-}
+	logrus.Printf("retrieved: %v \n", originalCM.SelfLink)
 
-func TestEditUnmanagedService(t *testing.T) {
-	client := testframework.MustNewClientset(t, nil)
-	defer testframework.MustManageConsole(t, client)
-	testframework.MustUnmanageConsole(t, client)
+	// copy for good measure
+	modifiedCM := originalCM.DeepCopy()
+	// modify it to check if it gets put back
+	modifiedCM.Data = map[string]string{}
+	modifiedCM.Data["this-is-not-the-right-yaml"] = string([]byte(`{"data": {"console-config.yaml": "test"}}`))
 
-	err := patchAndCheckService(t, client, false)
-	if err != nil {
+	// issue the update
+	if _, err := client.ConfigMaps(api.TargetNamespace).Update(modifiedCM); err != nil {
 		t.Fatalf("error: %s", err)
 	}
-}
 
-func TestEditUnmanagedRoute(t *testing.T) {
-	client := testframework.MustNewClientset(t, nil)
-	defer testframework.MustManageConsole(t, client)
-	testframework.MustUnmanageConsole(t, client)
+	// using this bool to test if the operator stomps the cm back into place
+	operatorHasReconciled := false
+	count := 0
+	err = wait.Poll(frequency, timeout, func() (stop bool, err error) {
+		count++
+		logrus.Printf("Polling: %v \n", count)
 
-	err := patchAndCheckRoute(t, client, false)
+		testCM, err := client.ConfigMaps(api.TargetNamespace).Get(api.OpenShiftConsoleConfigMapName, metav1.GetOptions{})
+		if err != nil {
+			// if we err, stop the loop, something went wrong
+			return true, err
+		}
+		logrus.Printf("cm Data: %v matches? %v \n", testCM.Data, !reflect.DeepEqual(originalCM.Data, testCM.Data))
```

Comment:
For the `Poll` method will return `err` when the timeout is reached `timed out waiting for the condition`., so we need to add a condition there that when the `count`, reaches the timeout value it will return without error. Thats why it was failing.

```go
...
	logrus.Printf("cm Data: %v matches? %v \n", testCM.Data, reflect.DeepEqual(originalCM.Data, testCM.Data))
	// this should stay false. if it flips to true we are broken.
	operatorHasReconciled = reflect.DeepEqual(originalCM.Data, testCM.Data)
	if count == 10 {
		return !operatorHasReconciled, nil
	}
	// let it run the full 10 cycles
	return operatorHasReconciled, err
})
...
```